src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -213,6 +225,52 @@ def run(
                 )
                 raise unexpected_error from e
 
+    def _copy_previous_run_result(
+        self,
+        resume_from_run_storage: AbstractBatchRunStorage,
+        resume_from_run_output_dir: Path,
+        batch_inputs: List,
+        output_dir: Path,
+    ) -> List[LineResult]:
+        """Duplicate the previous debug_info from resume_from_run_storage and output from resume_from_run_output_dir
+        to the storage of new run,
+        return the list of previous line results for the usage of aggregation and summarization.
+        """
+        previous_run_results = []
+        previous_run_output = (
+            load_list_from_jsonl(resume_from_run_output_dir / "output.jsonl") if resume_from_run_output_dir else []
+        )
+        previous_run_output_dict = {
+            each_line_output[LINE_NUMBER_KEY]: each_line_output for each_line_output in previous_run_output
+        }
+
+        if resume_from_run_output_dir:
+            copy_file_except(resume_from_run_output_dir, output_dir, "output.jsonl")
+
+        for i in range(len(batch_inputs)):
```

Comment:
Since the whole process is complicated, we may add more comments to explain what we are doing in each part.